code/turf/turf_flock.dm

```diff
@@ -356,6 +356,19 @@
 		else if(!isfeathertile(newloc))
 			F.end_floorrunning()
 
+/turf/simulated/wall/auto/feather/Bumped(AM)
+	. = ..()
+	if(istype(AM, /mob/living/critter/flock/drone))
+		var/mob/living/critter/flock/drone/F = AM
+		if(F.floorrunning || (F.can_floorrun && F.resources >= 1))
```

Comment:
```suggestion
		if((F.floorrunning || F.can_floorrun) && F.resources >= 1)
```
Shouldn't be able to floorrun with < 1 resource, but otherwise seems like saying you can?

Reply:
`can_floorrun` basically just acts as a flag for whether you're in a grab or not. So this if statement is really: "Are you floorrunning already, or are you not grabbed and have resources to start floorrunning?"